cas/cas-server/README.md

```diff
@@ -147,14 +147,280 @@ https://dev.vitamui.com:8080/cas/login).
+            ```yaml
+              cas.authn.accept.users: myusernam@mydomainmail.fr::mypassword
+              cas.authn.attribute-repository.stub.attributes.email: myusernam@mydomainmail.fr
+             ```
+          - **myusernam@mydomainmail.fr**  is the user email for testing, and **mypassword** is the password
+
+      - run ```mvn clean package``` on the project.
+      - run ```java -jar target/cas.war``` 
+      - After launching,cas will generate some settings files inside the directory ```/etc/cas/saml```
+      - Goto Vitamui, create a Saml provider with the following informations:
+        - Pattern: email domain configured before: ```mydomainmail.fr```
+        - Type: ```SAML```
+        - Email attribute: empty (except if the cas send the email after authentication)
+        - CAS Keystore: for testing: you upload any keystore with the right passowrd.
+        - IDP Metadata: upload the file generated by running external CAS, from the path ```/etc/cas/saml/idp-metadata.xml```
+        - Assertions : false
+        - Signed request: false.
+      - On vitamui, after creating the provider (SSO list) : 
+        - we download the ```SPS-metadata.xml``` file and copy it in the vm of external CAS, in any directory, 
+            example ```/vitam/some-path-of-cas/SPS-metadata.xml```. 
+        - create a new resource file on the project external CAS , example: ```saml-{{my-env-ui}}.json``` in the directory:
+             src/main/resources/services with the following content:
+        ```json
+           {
+          "@class" : "org.apereo.cas.support.saml.services.SamlRegisteredService",
+          "name" : "SAMLService",
+          "id" : 1,
+          "evaluationOrder" : 1,
+          "metadataLocation" : "/vitam/some-path-of-cas/SPS-metadata.xml",
+          "skipGeneratingTransientNameId": true,
+          "serviceId" : "https://{{my-env-ui}}.env.programmevitam.fr/cas/login/{{technical-provider-id}}"
+          }
+        ``` 
+        We have to check the following important informations: 
+        **my-env-ui** is the name of test environment ex: int.
+        **some-path-of-cas** is a directory path on the CAS external path vm.
+        **technical-provider-id** is the ```technicalName``` of the provider in providers collection in vitamui db created before.
+    - run ```mvn clean package``` on the project.
+    - run ```java -jar target/cas.war```
+
+- You need to restart the CAS service on the test environment (ex: INT), after adding the provider.
+- Create a user into the organisation with email address ```myusernam@mydomainmail.fr``` 
+- We can test the SAML delegated authentication using the external CAS. 
+      
+
+## Auto provisionning:
+
+The auto provisioning is a feature enabling either the updating or creation of users following authentication from an external IdP.
+It calls an API to develop to provision a user based on their primary email address, which is the main authentication information. 
```

Comment:
When enabled, the auto-provisioning of user calls an ad-hoc back-end API that returns user information based on their primary email address.